src/FSharp.Core/prim-types.fs

```diff
@@ -3912,6 +3912,14 @@ namespace Microsoft.FSharp.Collections
     type List<'T> = 
        | ([])  :                  'T list
        | ( :: )  : Head: 'T * Tail: 'T list -> 'T list
+       member private this.CustomHashCode(c:IEqualityComparer) =        
+            let rec loop l acc position =
+                match l with
+                | [] -> acc           
+                | h::t ->
+                    loop t  (LanguagePrimitives.HashCompare.HashCombine position acc (c.GetHashCode(h))) (position+1)
```

Comment:
I'm pretty sure the call to `c.GetHashCode(h)` should be instead `GenericHashWithComparer c h`.  This is what we were code generating: https://github.com/dotnet/fsharp/blob/d796a1ce9b6ccdc5d91b1594bd701ccbebb7834d/src/Compiler/Checking/AugmentWithHashCompare.fs#L628

I know array hashing doesn't call this, but we should do it to keep the semantics precisely the same - it looks like there are edge cases in limited hashing where this will matter. 

Also if we ever decide to inline `CustomHashCode` for lists into user code then the right inline optimizations will kick in (defunctionalizing and flattening the whole hash code for, say `(int * int) list` )

Reply:
Did it.

Also added a tests for ["dada";null;null] because our GenericHashWithComparer covers for it, but a comparer implementation could be throwing an ArgumentNullException.

Reply:
I can't spot any significant uses of `LimitedStructural`, e.g. [this search](https://www.google.com/search?rlz=1C1ONGR_en-GBGB970GB970&q=%22LimitedStructural%22&nfpr=1&sa=X&ved=2ahUKEwi_9OuAzbD8AhWAQkEAHeNfBKEQvgUoAXoECDcQAg&biw=1920&bih=1081&dpr=1).

This feature was present in F# 1.0 and was quite prominent then, but we gradually made it less prominent and it should have just been deleted in F# 2.0.  Let's just deprecate it completely now.

Reply:
@T-Gro Please do that in a separate PR.

Reply:
This seems to be the only usage I can find which is not a fork of F# or it's docs:

https://github.com/mrakgr/The-Spiral-Language/commit/82689073af9c993734dad97f60714f4d39e41154 

Will raise a separate deprecating PR later for it.

Reply:
Thanks!  Make a tracking language suggestion too please